app/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataRecord.java

```diff
@@ -2,184 +2,151 @@
 
 import org.sagebionetworks.bridge.json.DateTimeJsonDeserializer;
 import org.sagebionetworks.bridge.json.DateTimeJsonSerializer;
-import org.sagebionetworks.bridge.json.JsonUtils;
+import org.sagebionetworks.bridge.json.LocalDateToStringSerializer;
 import org.sagebionetworks.bridge.models.healthdata.HealthDataRecord;
+import org.sagebionetworks.bridge.models.healthdata.HealthDataRecordBuilder;
 
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexHashKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.datatype.joda.deser.LocalDateDeserializer;
+import org.joda.time.LocalDate;
 
-@DynamoDBTable(tableName = "HealthDataRecord2")
+/** DynamoDB implementation of {@link org.sagebionetworks.bridge.models.healthdata.HealthDataRecord}. */
+@DynamoDBTable(tableName = "HealthDataRecord3")
 public class DynamoHealthDataRecord implements HealthDataRecord {
-
-    private static final String GUID_FIELD = "guid";
-    private static final String START_DATE_FIELD = "startDate";
-    private static final String END_DATE_FIELD = "endDate";
-    private static final String VERSION_FIELD = "version";
-    private static final String DATA_FIELD = "data";
-    
-    private String key;
-    private String guid;
-    private long startDate;
-    private long endDate;
+    private Long createdOn;
     private JsonNode data;
+    private String healthCode;
+    private String id;
+    private JsonNode metadata;
+    private String schemaId;
+    private LocalDate uploadDate;
```

Comment:
What's the thinking behind having uploadDate a LocalDate? It would seem that we know the time in an absolute sense, wouldn't it be a Long?